src/sensors/linearServo/LinearServo.java

```diff
@@ -32,18 +32,51 @@ public LinearServo(int channel, int maxLength, int maxSpeed) {
     }
 
     /**
+     * convert setpoint[mm] to PWM value
+     *
      * @param setpoint the target position of the servo [mm]
      */
-    public void setPosition(double setpoint) {
+    private double convertSetpointToPWM(double setpoint){
         setPos = MathUtil.clamp(setpoint, 0, maxLength);
-        setSpeed((setPos / maxLength * 2) - 1);
+        return (setPos / maxLength * 2) - 1;
+    }
+
+    /**
+     * @param setpoint the target position of the servo [mm]
+     */
+    public void setPosition(double setpoint) {
+        super.setSpeed(convertSetpointToPWM(setpoint));
+    }
+
+    /** calibrate the linear servo
+     * you can override this function and use different position or use setPosition once.
+     *
+     * the calibrate position doesn't meter and you only need to do this once*/
+    public void calibrate() {
+        setPosition(0);
+    }
+
+    /** move the linear servo forward or reverse according to the speed sign
+     *
+     *@param speed determine the direction
+     *
+     * speed equal to 0 stops the linear servo
+     * speed > 0 moves the linear servo forward
+     * speed < 0 moves the linear servo reverse
+     *
+     * dont forget to calibrate once the linearServo other wise it will not stop.
+     * */
+    @Override
+    public void setSpeed(double speed) {
+        setPosition(speed == 0 ? currentPos : Math.min(Math.max(Math.ceil(speed),0) * maxLength , maxLength));
```

Comment:
You dont need Math.min,,

Reply:
It prevent from getting set point bigger from maxLength